batcher.js

```diff
@@ -13,22 +13,22 @@ if (process.argv.length < 4) {
 
 var command = process.argv[2];
 var query = process.argv[3];
-var data = parseData(process.argv[4]);
+var searchParams = sToO(process.argv[4]);
+var data = sToO(process.argv[5]);
 
-function parseData(data) {
-    // FIXME: not particularly typesafe.
-    try {
-        return JSON.parse(data)
-    } catch(e) {
-        return undefined;
+function sToO(s) {
+    // Wow!
+    eval('const o = '+ s);
```

Comment:
This code is vulnerable to [an injection attack](https://www.owasp.org/index.php/Top_10_2013-A1-Injection), which is at the top of [OWASP's top 10](https://www.owasp.org/index.php/Top_10_2013-Top_10). You're evaluating an unverified string that comes from outside the program, consider `s = "1; <any valid JavaScript here will be run in that scope>";`.

This is only a local script or whatevs so it's up to you all, but I'd recommend:

a) not getting into the habit
b) not putting code like that into repos that are shared with other devs who may not know better
c) just never doing that

It's totally up to you all but if you decide this is ok please add a comment saying you realise it is a bit silly but it's ok for "these reasons" so people don't copy it elsewhere and there's a chance it'll get done differently if this script ever starts being used in a more vulnerable context.